File: 16-3sum-closest/16-3sum-closest.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {
        Arrays.sort(nums);
        int sum = nums[0] + nums[1] + nums[2];
        for (int i = 0; i < nums.length - 2; i++) {
            int left = i + 1, right = nums.length - 1;
            while (left <right) {
                int currSum = nums[i] + nums[left] + nums[right];
                if (Math.abs(currSum - target) < Math.abs(sum - target)) {
                    sum = currSum;
                }
                if (currSum < target) left++;
                else if (currSum > target) right--;
                else return sum;
            }
        }
        return sum;
    }
}

File: 16-3sum-closest/16-3sum-closest.py
class Solution(object):
    def threeSumClosest(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        nums.sort()
        sum = nums[0] + nums[1] + nums[2]
        for i in range(len(nums) - 2):
            left, right = i + 1, len(nums) - 1
            while left < right:
                currSum = nums[i] + nums[left] + nums[right]
                if abs(currSum - target) < abs(sum - target):
                    sum = currSum
                if currSum < target: left += 1
                elif currSum > target: right -= 1
                else: return sum
        return sum